src/session.js

```diff
@@ -0,0 +1,38 @@
+let crypto = require('crypto');
+
+let sessions = {};
+
+function hydrate(request, sessionId) {
+  let session = sessions[sessionId];
+  request.serial = session.serial;
+  request.remoteHost = session.remoteHost;
+  request.remotePort = session.remotePort;
+}
+
+function createHash(data) {
+  let hash = crypto.createHash('sha1');
+  hash.update(JSON.stringify(data));
+  return hash.digest('hex');
+}
+
+function middleware(req, res, next) {
+  let sessionId = req.get('Session-Id');
```

Comment:
Hmm, I had thought `Session-Id` was somewhat standard, but upon looking I cannot find it. I'll change to X prefix.